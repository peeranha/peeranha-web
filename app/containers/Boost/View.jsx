import React, { useCallback, useState } from 'react';
import PropTypes from 'prop-types';

import { MAX_STAKE_PREDICTION, MIN_STAKE_PREDICTION } from './constants';

import { getBoostWeeks } from 'utils/walletManagement';

import NavHeader from 'components/WalletNavigation';
import SubHeader from 'containers/Wallet/SubHeader';
import Weeks from './Weeks';
import Form from './Form';

const View = ({
  userId,
  locale,
  account,
  balance,
  stakedInCurrentPeriod,
  stakedInNextPeriod,
  weekStat,
  globalBoostStat,
  userBoostStat,
  getWeekStatProcessing,
  changeStakeDispatch,
  changeStakeLoading,
}) => {
  const boostWeeks = getBoostWeeks(weekStat, globalBoostStat, userBoostStat);
  const { currentWeek, nextWeek } = boostWeeks;
  const { userStake, maxStake } = nextWeek;

  const [currentStake, setCurrentStake] = useState(userStake);

  if (currentStake === undefined && !!userStake)
    setCurrentStake(userStake);

  const changeCurrentStake = useCallback(
    x => {
      setCurrentStake(Math.floor(balance * x * 1000000) / 1000000);
    },
    [currentStake, setCurrentStake],
  );

  let predictedBoost = 1;
  if (maxStake && currentStake <= maxStake) {
    predictedBoost = currentStake / maxStake * (MAX_STAKE_PREDICTION - MIN_STAKE_PREDICTION) + 1;
    predictedBoost = Math.floor(predictedBoost * 100) / 100;
  } else if (currentStake > 0) {
    predictedBoost = MAX_STAKE_PREDICTION;
  }

  return (
    <>
      <NavHeader userId={userId} />
      
      <SubHeader
        account={account}
        balance={balance ? balance.toString() : ""}
        stakedInCurrentPeriod={stakedInCurrentPeriod}
        stakedInNextPeriod={stakedInNextPeriod}
      />

      <Weeks
        locale={locale}
        weekStat={weekStat}
        globalBoostStat={globalBoostStat}
        userBoostStat={userBoostStat}
        getWeekStatProcessing={getWeekStatProcessing}
      />

      <Form
        currentStake={currentStake}
        predictedBoost={predictedBoost}
        maxStake={balance}
        initialUserStake={stakedInNextPeriod}
        changeCurrentStake={changeCurrentStake}
        onChangeCurrentStake={setCurrentStake}
        changeStake={changeStakeDispatch}
        changeStakeLoading={changeStakeLoading}
        locale={locale}
      />
    </>
  );
}

View.propTypes = {
  userId: PropTypes.string,
  locale: PropTypes.string,
  account: PropTypes.string,
  balance: PropTypes.number,
  stakedInCurrentPeriod: PropTypes.number,
  stakedInNextPeriod: PropTypes.number,
  weekStat: PropTypes.array,
  globalBoostStat: PropTypes.array,
  userBoostStat: PropTypes.array,
  getWeekStatProcessing: PropTypes.bool,
  changeStakeDispatch: PropTypes.func,
  changeStakeLoading: PropTypes.bool,
};

export default React.memo(View);
