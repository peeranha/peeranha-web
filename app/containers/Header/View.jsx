/**
 *
 * View
 *
 */

import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { injectIntl, intlShape, FormattedMessage } from 'react-intl';

import { BG_LIGHT, BORDER_SECONDARY } from 'style-constants';

import * as routes from 'routes-config';
import messages from 'common-messages';

import addIcon from 'images/add.svg?external';
import searchIcon from 'images/search.svg?inline';
import headerNavigationIcon from 'images/headerNavigation.svg?inline';
import img from 'images/LogoBlack.svg?inline';

import LargeButton from 'components/Button/Contained/InfoLarge';
import Icon from 'components/Icon';

import Wrapper from './Wrapper';
import Section from './Section';
import Logo from './Logo';

import ButtonGroupForNotAuthorizedUser from './ButtonGroupForNotAuthorizedUser';
import ButtonGroupForAuthorizedUser from './ButtonGroupForAuthorizedUser';
import SearchForm from './SearchForm';

import { HEADER_ID } from './constants';

export const LoginProfile = React.memo(
  ({ profileInfo, showLoginModalDispatch }) => {
    if (profileInfo) {
      return <ButtonGroupForAuthorizedUser profileInfo={profileInfo} />;
    }

    return (
      <ButtonGroupForNotAuthorizedUser
        showLoginModal={showLoginModalDispatch}
      />
    );
  },
);

const Button = LargeButton.extend`
  background-color: ${x => x.bg};
  border: ${x => (x.bg ? '1' : '0')}px solid ${BORDER_SECONDARY};

  @media only screen and (max-width: 992px) {
    padding: 0;
    border-radius: 50%;
    min-width: auto;
    width: 42px;
    height: 42px;
  }

  @media only screen and (max-width: 576px) {
    width: 36px;
    height: 36px;
  }
`;

const View = ({
  showMenu,
  intl,
  profileInfo,
  showLoginModalDispatch,
  redirectToAskQuestionPage,
}) => {
  const [isSearchFormVisible, setSearchFormVisibility] = useState(false);
  const searchFormId = 'q';

  useEffect(
    () => {
      if (isSearchFormVisible) {
        document.getElementById(searchFormId).focus();
      }
    },
    [isSearchFormVisible],
  );

  return (
    <Wrapper id={HEADER_ID}>
      <div className="container">
        <div className="d-flex align-items-center justify-content-between">
          <div className="d-flex align-items-center">
            <button className="mt-1 mr-3 d-flex d-md-none" onClick={showMenu}>
              <img src={headerNavigationIcon} alt="switch" />
            </button>

            {!isSearchFormVisible && (
              <Logo to={routes.questions()}>
                <img src={img} alt="logo" />
              </Logo>
            )}
          </div>

          <Section>
            <SearchForm
              searchFormId={searchFormId}
              onBlur={() => setSearchFormVisibility(false)}
              className={`${isSearchFormVisible ? '' : 'd-none'} d-lg-flex`}
              placeholder={intl.formatMessage({
                id: messages.search.id,
              })}
            />

            {!isSearchFormVisible && (
              <Button
                bg={BG_LIGHT}
                className="d-flex d-lg-none"
                onClick={() => setSearchFormVisibility(!isSearchFormVisible)}
              >
                <img src={searchIcon} alt="icon" />
              </Button>
            )}

            {!isSearchFormVisible && (
              <Button
                id="header-ask-question"
                onClick={redirectToAskQuestionPage}
              >
                <Icon color={BG_LIGHT} icon={addIcon} width="14" noMargin />
                <span className="d-none d-lg-inline ml-2">
                  <FormattedMessage {...messages.addQuestion} />
                </span>
              </Button>
            )}

            <LoginProfile
              showLoginModalDispatch={showLoginModalDispatch}
              profileInfo={profileInfo}
            />
          </Section>
        </div>
      </div>
    </Wrapper>
  );
};

View.propTypes = {
  intl: intlShape.isRequired,
  profileInfo: PropTypes.object,
  isMenuVisible: PropTypes.bool,
  showMenu: PropTypes.func,
  showLoginModalDispatch: PropTypes.func,
  redirectToAskQuestionPage: PropTypes.func,
};

LoginProfile.propTypes = {
  profileInfo: PropTypes.object,
  showLoginModalDispatch: PropTypes.func,
};

export default injectIntl(React.memo(View));
