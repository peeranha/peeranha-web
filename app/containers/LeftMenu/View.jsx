import React from 'react';
import PropTypes from 'prop-types';

import { LEFT_MENU_ID } from './constants';

import AdditionalLinks from './AdditionalLinks';
import MobileLinksInWallet from './MobileLinksInWallet';
import MainLinks from './MainLinks';
import MobileLinksInProfile from './MobileLinksInProfile';
import MobileAdditionalLinks from './MobileAdditionalLinks';
import MobileAutorizationButtons from './MobileAutorizationButtons';
import { ViewStyled } from './Styles';

const View = ({ profile, isMenuVisible, balance, showLoginModal }) => (
  <ViewStyled id={LEFT_MENU_ID} isMenuVisible={isMenuVisible}>
    <MobileAutorizationButtons
      profile={profile}
      isMenuVisible={isMenuVisible}
      showLoginModal={showLoginModal}
    />

    <MobileLinksInProfile profile={profile} isMenuVisible={isMenuVisible} />

    <MobileLinksInWallet
      profile={profile}
      isMenuVisible={isMenuVisible}
      balance={balance}
    />

    <MobileAdditionalLinks profile={profile} isMenuVisible={isMenuVisible} />

    <MainLinks profile={profile} />

    <AdditionalLinks />
  </ViewStyled>
);

View.propTypes = {
  profile: PropTypes.object,
  balance: PropTypes.number,
  isMenuVisible: PropTypes.bool,
  showLoginModal: PropTypes.func,
};

export default View;
