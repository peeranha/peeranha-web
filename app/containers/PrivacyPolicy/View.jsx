import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import * as routes from 'routes-config';
import commonMessages from 'common-messages';

import { TEXT_PRIMARY } from 'style-constants';

import { getIndexes, getQuestionCode } from 'utils/privacyPolicyManagement';

import A from 'components/A';
import H3 from 'components/H3';
import Base from 'components/Base/BaseRounded';
import BaseTransparent from 'components/Base/BaseTransparent';
import { TextBlock } from 'containers/Faq/Content';
import { BaseStyled } from 'components/TextEditor/Tips';

const NavItem = A.extend`
  display: block;
  padding: 6px 0;

  :before {
    font-weight: 600;
    padding-right: 5px;
    color: ${TEXT_PRIMARY};
    content: '${x => x.index + 1}.';
  }
`.withComponent('a');

const View = ({ privacyPolicy }) => {
  const [sectionIndex, questionIndex] = getIndexes(window.location.hash);

  if (!privacyPolicy) {
    return null;
  }

  const section = privacyPolicy.blocks[sectionIndex];
  const question = section.blocks[questionIndex];

  return (
    <div>
      <Base className="mb-3">
        <H3>
          <FormattedMessage {...commonMessages.privacyPolicy} />
        </H3>
      </Base>

      <Base className="p-0">
        <div className="d-flex">
          <div className="col-12 col-xl-9 p-0">
            <BaseTransparent>
              <TextBlock
                isOpened
                dangerouslySetInnerHTML={{ __html: question.content }}
              />
            </BaseTransparent>
          </div>

          <div className="col-12 col-xl-3 p-0">
            <BaseStyled>
              {section.blocks.map((x, index) => (
                <NavItem
                  index={index}
                  href={routes.privacyPolicy(
                    getQuestionCode(sectionIndex, index),
                  )}
                >
                  {x.h3}
                </NavItem>
              ))}
            </BaseStyled>
          </div>
        </div>
      </Base>
    </div>
  );
};

View.propTypes = {
  privacyPolicy: PropTypes.array,
};

export default View;
