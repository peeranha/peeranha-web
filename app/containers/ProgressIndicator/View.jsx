import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';

import { BG_PRIMARY_LIGHT, BORDER_SECONDARY } from 'style-constants';

import processIndicator from 'images/progress-indicator.svg?inline';

import {
  HEADER_HEIGHT,
  MOBILE_HEADER_HEIGHT,
} from 'containers/Header/constants';

const width = 156;
const height = 50;

const Box = styled.div`
  position: fixed;
  top: ${x => (x.inProgress ? HEADER_HEIGHT * 1.25 : -HEADER_HEIGHT)}px;
  opacity: ${x => (x.inProgress ? 1 : 0)};
  left: calc(50% - ${width / 2}px);
  width: ${width}px;
  height: ${height}px;
  z-index: 1000000;
  transition: 0.5s;
  border-radius: 5px;
  box-shadow: 0 2px 2px 0 ${BORDER_SECONDARY};
  background-color: ${BG_PRIMARY_LIGHT};
  display: flex;
  justify-content: center;
  align-items: center;

  img {
    margin-right: 10px;
    animation: rotation 1s infinite linear;
  }

  span:after {
    content: '...';
  }

  @media only screen and (max-width: 991px) {
    top: ${x =>
      x.inProgress ? MOBILE_HEADER_HEIGHT * 1.25 : -MOBILE_HEADER_HEIGHT}px;
  }

  @keyframes rotation {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;

const View = ({ inProgress }) => {
  const { t } = useTranslation();

  return (
    <Box inProgress={inProgress}>
      <img src={processIndicator} alt="icon" />
      {t('common.inProgress')}
    </Box>
  );
};

View.propTypes = {
  inProgress: PropTypes.bool,
};

export default React.memo(View);
