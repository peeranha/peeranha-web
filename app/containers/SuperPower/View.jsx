import React from 'react';
import PropTypes from 'prop-types';

import NavHeader from 'components/WalletNavigation';
import SubHeader from 'containers/Wallet/SubHeader';
import Weeks from './Weeks';

const View = ({
  userId,
  locale,
  account,
  balance,
  weekStat,
  getWeekStatProcessing,
}) => (
  <>
    <NavHeader userId={userId} />
    
    <SubHeader account={account} balance={balance} />

    <Weeks
      locale={locale}
      weekStat={weekStat}
      getWeekStatProcessing={getWeekStatProcessing}
    />
  </>
);

View.propTypes = {
  userId: PropTypes.string,
  locale: PropTypes.string,
  account: PropTypes.string,
  balance: PropTypes.number,
  weekStat: PropTypes.array,
  getWeekStatProcessing: PropTypes.bool,
};

export default React.memo(View);
