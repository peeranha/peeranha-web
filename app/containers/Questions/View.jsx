import React from 'react';
import PropTypes from 'prop-types';
import * as routes from 'routes-config';

import LoadingIndicator from 'components/LoadingIndicator/WidthCentered';
import TopCommunities from 'components/TopCommunities';

import Header from './Header';
import Content from './Content';

const feed = routes.feed();

/* eslint indent: 0 */
export const View = props => (
  <React.Fragment>
    <Header {...props} />

    {props.parentPage === feed && (
      <TopCommunities
        userId={props.account}
        account={props.account}
        communities={props.communities}
        profile={props.profile}
      />
    )}

    <Content {...props} />

    {(props.questionsLoading || props.communitiesLoading) && (
      <LoadingIndicator />
    )}
  </React.Fragment>
);

View.propTypes = {
  questionsList: PropTypes.array,
  followedCommunities: PropTypes.array,
  communities: PropTypes.array,
  questionsLoading: PropTypes.bool,
  communitiesLoading: PropTypes.bool,
  parentPage: PropTypes.string,
  account: PropTypes.string,
  profile: PropTypes.object,
};

export default React.memo(View);
