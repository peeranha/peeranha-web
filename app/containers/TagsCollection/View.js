import React from 'react';
import PropTypes from 'prop-types';

import Header from './Header';
import List from './List';

const View = ({ communities }) => (
  <React.Fragment>
    <Header />
    <List communities={communities} />
  </React.Fragment>
);

View.propTypes = {
  communities: PropTypes.array,
};

export default React.memo(View);
