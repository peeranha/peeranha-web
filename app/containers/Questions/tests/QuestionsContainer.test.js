import { View } from '../View';

const props = {
  translations: {},
  questionsList: [],
  questionsLoading: false,
  locale: 'en',
};

describe('View', () => {
  it('snapshot test 1', () => {
    expect(View(props)).toMatchSnapshot();
  });

  it('snapshot test 2', () => {
    props.questionsList = ['cheburek'];
    expect(View(props)).toMatchSnapshot();
  });

  it('snapshot test 3', () => {
    props.questionsLoading = true;
    expect(View(props)).toMatchSnapshot();
  });
});
