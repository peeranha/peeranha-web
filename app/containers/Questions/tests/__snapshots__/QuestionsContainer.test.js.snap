// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionsContainer snapshot test 1 1`] = `
<div>
  <InjectIntl(Component)
    locale="en"
    questionsList={Array []}
    questionsLoading={false}
    translations={Object {}}
  />
  <UNDEFINED
    isFeed={false}
  />
  <UNDEFINED
    locale="en"
    questionsList={Array []}
    questionsLoading={false}
    translations={Object {}}
  />
</div>
`;

exports[`QuestionsContainer snapshot test 2 1`] = `
<div>
  <InjectIntl(Component)
    locale="en"
    questionsList={
      Array [
        "cheburek",
      ]
    }
    questionsLoading={false}
    translations={Object {}}
  />
  <UNDEFINED
    locale="en"
    questionsList={
      Array [
        "cheburek",
      ]
    }
    questionsLoading={false}
    translations={Object {}}
  />
</div>
`;

exports[`QuestionsContainer snapshot test 3 1`] = `
<div>
  <InjectIntl(Component)
    locale="en"
    questionsList={
      Array [
        "cheburek",
      ]
    }
    questionsLoading={true}
    translations={Object {}}
  />
  <UNDEFINED
    locale="en"
    questionsList={
      Array [
        "cheburek",
      ]
    }
    questionsLoading={true}
    translations={Object {}}
  />
  <UNDEFINED />
</div>
`;
